apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/RewriteStrategyGeneratorSparkApp.java

```diff
@@ -0,0 +1,54 @@
+package com.linkedin.openhouse.jobs.spark;
+
+import com.linkedin.openhouse.datalayout.datasource.TableFileStats;
+import com.linkedin.openhouse.datalayout.datasource.TablePartitionStats;
+import com.linkedin.openhouse.datalayout.generator.OpenHouseRewriteStrategyGenerator;
+import com.linkedin.openhouse.datalayout.persistence.StrategiesDao;
+import com.linkedin.openhouse.datalayout.persistence.StrategiesDaoTableProps;
+import com.linkedin.openhouse.datalayout.strategy.RewriteStrategy;
+import com.linkedin.openhouse.jobs.spark.state.StateManager;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.spark.sql.SparkSession;
+
+@Slf4j
+public class RewriteStrategyGeneratorSparkApp extends BaseTableSparkApp {
```

Comment:
DataLayoutStrategyGeneratorSparkApp

Reply:
It was long `DataLayoutOptimizationStrategyGeneratorSparkApp` :) `DataLayoutStrategy` is missing action like rewrite, or optimization. What would you suggest, do you want `DataLayout` to be part of the name?

Reply:
Yeah rewrite strategy is too generic

Reply:
Fine with DataLayoutStrategyGenerator